packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1061,11 +1065,12 @@ export class TransactionController extends BaseController<
   ): TransactionMeta[] {
     const nonceNetworkSet = new Set();
     const txsToKeep = transactions.reverse().filter((tx) => {
-      const { chainId, networkID, status, transaction, time } = tx;
+      const { caipChainId, networkID, status, transaction, time } = tx;
       if (transaction) {
-        const key = `${transaction.nonce}-${
-          chainId ? convertHexToDecimal(chainId) : networkID
-        }-${new Date(time).toDateString()}`;
+        const networkChainId = caipChainId || networkID; // is this right?
```

Comment:
It might be good to keep the same behavior and convert the CAIP chain ID to decimal so that keys for existing transactions are the same. Otherwise transactions that are already in state might get accidentally dropped.

Reply:
(Side note: it might be good to ask these questions in GitHub comments rather than code comments so they don't get accidentally committed.)